infra/helper.py

```diff
@@ -1703,13 +1712,28 @@ def index(args):
 
   run_args = _env_to_docker_args(env)
   run_args.extend([
-      '-v', f'{args.project.out}:/out', '-v', f'{args.project.work}:/work',
-      '-t', image_name, '/opt/indexer/index_build.py'
+      '-v',
+      f'{args.project.out}:/out',
+      '-v',
+      f'{args.project.work}:/work',
+      '-t',
   ])
 
+  if args.docker_args:
+    run_args.extend(shlex.split(args.docker_args))
+
+  run_args.append(image_name)
+  if args.shell:
+    run_args.append('/bin/bash')
+  else:
+    run_args.append('/opt/indexer/index_build.py')
+
   if args.targets:
     run_args.extend(['--targets', args.targets])
 
+  if args.extra_args:
```

Comment:
There is probably no strict need for the `if` check here, this should be an empty list by default.

Reply:
done!